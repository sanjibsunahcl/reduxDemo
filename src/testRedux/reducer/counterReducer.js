import {createSlice} from '@reduxjs/toolkit';

export const counterSlice = createSlice({
  name: 'counter',
  initialState: {
    value: 0,
    increment: 1,
  },
  reducers: {
    increment: state => {
      state.value += state.increment;
    },
    decrement: state => {
      state.value -= state.increment;
    },
    updateIncrement: (state, action) => {
      state.increment = action.payload;
    },
  },
});

// Action creators are generated for each case reducer function
export const {increment, decrement, updateIncrement} = counterSlice.actions;

export default counterSlice.reducer;
