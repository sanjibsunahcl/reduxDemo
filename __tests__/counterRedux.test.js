import React from 'react';
import CounterTest from '../src/screen/reduxTest/Counter';
import {Provider, connect} from 'react-redux';
import {render, fireEvent} from '@testing-library/react-native';
import {configureStore} from '@reduxjs/toolkit';
import counterReducer from '../src/testRedux/reducer/counterReducer';

function renderWithRedux(renderComponent) {
  const store = configureStore({
    reducer: {
      counter: counterReducer,
    },
  });

  return render(<Provider store={store}>{renderComponent}</Provider>);
}

describe('Testing counter page', () => {
  it('renders the component correctly', () => {
    renderWithRedux(<CounterTest />);
  });

  it('Should start with the current count of 0', () => {
    const {getByText, getByTestId} = renderWithRedux(<CounterTest />);

    fireEvent.press(getByText('+'));
    console.log('fjhdfhgjh' + getByTestId('count-value').props.children);
    expect(getByTestId('count-value').props.children).toBe(1);

    // const count = getByText('0');

    // expect(count).not.toBeNull();
  });
});
