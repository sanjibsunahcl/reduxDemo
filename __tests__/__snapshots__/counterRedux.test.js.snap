// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing counter page Should start with the current count of 0 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "#3D2C8D",
        "fontSize": 25,
        "fontWeight": "bold",
        "marginBottom": 20,
      }
    }
  >
    Redux Test
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#3D2C8D",
          "borderRadius": 60,
          "height": 60,
          "justifyContent": "center",
          "opacity": 1,
          "width": 60,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 28,
          }
        }
      >
        -
      </Text>
    </View>
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 20,
          "marginHorizontal": 30,
        }
      }
      testID="count-value"
    >
      0
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#3D2C8D",
          "borderRadius": 60,
          "height": 60,
          "justifyContent": "center",
          "opacity": 1,
          "width": 60,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 28,
          }
        }
      >
        +
      </Text>
    </View>
  </View>
</View>
`;
